### ISSUE TITLE
Training Logs

### ISSUE THREAD

**Issue by steven850** [OlaWod/FreeVC#25]
Hi, wondering if you could post your logs from the model trained to 900k steps? Would love to take a look at the progression of the losses during training etc.

**Comment by OlaWod** [comment]
Sure. 
Red: freevc (using `data_utils_old.py`)
Pink: freevc-s  (using `data_utils_old.py`)
Orange: freevc  (using `data_utils.py`)

![IMG_2103](https://user-images.githubusercontent.com/32514635/210977209-ecda8b60-0263-44c2-9bc8-635e8e4acdcf.PNG)
![IMG_2104](https://user-images.githubusercontent.com/32514635/210977218-aeec409f-b4a7-4789-855d-fbf21b0c94ea.PNG)
![IMG_2105](https://user-images.githubusercontent.com/32514635/210977222-7e9f9fe4-fab0-4ca5-a33e-dcd5abd966ab.PNG)
![IMG_2097](https://user-images.githubusercontent.com/32514635/210977227-6870aa57-bb9f-451b-a0bd-bd799e1a7464.PNG)
![IMG_2098](https://user-images.githubusercontent.com/32514635/210977231-c7443e61-8532-4455-af3e-c268703ab100.PNG)
![IMG_2099(20230106-175042)](https://user-images.githubusercontent.com/32514635/210977234-82223fdd-bc2a-4856-b221-7cbecb080fc5.PNG)
![IMG_2100](https://user-images.githubusercontent.com/32514635/210977236-ca807b9b-1861-44c0-b2d4-ec752ccaafb1.PNG)
![IMG_2106](https://user-images.githubusercontent.com/32514635/210977224-64b9a09f-f303-42b6-b1ab-6bfa7ee500f3.PNG)
![IMG_2102(20230106-175250)](https://user-images.githubusercontent.com/32514635/210977201-6aa9dedc-276f-4e04-8a03-793c40ea8e1f.PNG)

**Comment by steven850** [comment]
Thanks for posting these. 
Why does your loss/g/total and loss/g/fm climb?  
mine decrease over the training run. 
does this suggest a problem on my end
![gitloss](https://user-images.githubusercontent.com/43098060/211349557-46a422e1-e564-42a3-ae1d-4c32461fadbf.PNG)
?

**Comment by steven850** [comment]
My generated audio is coming out ok on training eval. Cleaning up as it goes, but I have noticed that the the actual conversion output of the model seems to be getting worse, comparing 245, 345, and 480K the output quality is getting worse. So the 245k model is very similar to target, and the content is clean. The 480k in comparison, has a lot of distortion, the content is less intelligible, and the pitch is off.

**Comment by OlaWod** [comment]
The code had some problem, I fixed it yesterday. Sorry to have wasted your time on problematic code. 😥

**Comment by OlaWod** [comment]
> I will try the new code and report back, thank you for the update.

Hmmm I found I forgot to de comment [this line](https://github.com/OlaWod/FreeVC/blob/main/train.py#L131) just now... But it might not be a big problem I think. And thank you for your interest in our work!

**Comment by skol101** [comment]
> > I will try the new code and report back, thank you for the update.
> 
> Hmmm I found I forgot to de comment [this line](https://github.com/OlaWod/FreeVC/blob/main/train.py#L131) just now... But it might not be a big problem I think. And thank you for your interest in our work!

Show this line be commented out?

**Comment by OlaWod** [comment]
> > > I will try the new code and report back, thank you for the update.
> > 
> > 
> > Hmmm I found I forgot to de comment [this line](https://github.com/OlaWod/FreeVC/blob/main/train.py#L131) just now... But it might not be a big problem I think. And thank you for your interest in our work!
> 
> Show this line be commented out?

Yes it need to be commented out. (I don't even remember why I commented this line before uploading code...) 
As in [pytorch documentation](https://pytorch.org/docs/stable/data.html?highlight=set_epoch):
> In distributed mode, calling the set_epoch() method at the beginning of each epoch before creating the DataLoader iterator is necessary to make shuffling work properly across multiple epochs. Otherwise, the same ordering will be always used.

**Comment by OlaWod** [comment]
> Does distributed mode relate to the case when gpus > 1

This code uses distributed mode no matter how many gpus used. No sure if `set_epoch` only affects distributed mode when gpus > 1 or all distributed mode settings.

**Comment by steven850** [comment]
Trying to resume the original model with the code changes you made. On previous runs the loss trend matched your original logs as far as increases and decreases goes. With the new code I am seeing a drop in FM loss, is this to be expected or a sign of a problem? 
![fmloss](https://user-images.githubusercontent.com/43098060/212043420-060e6018-d244-47fc-a95e-1d6ac243b961.PNG)

**Comment by OlaWod** [comment]
This [vits issue](https://github.com/jaywalnut310/vits/issues/14#issuecomment-1010287017) is a good reference for your question.
